src/Sprite.js

```diff
@@ -481,6 +481,60 @@ export class Sprite extends SpriteBase {
     } while (t < 1);
   }
 
+  ifOnEdgeBounce() {
+    const nearestEdge = this.nearestEdge();
+    if (!nearestEdge) return;
+    const rad = this.scratchToRad(this.direction);
+    let dx = Math.cos(rad);
+    let dy = Math.sin(rad);
+    switch (nearestEdge) {
+      case Sprite.Edge.LEFT:
+        dx = Math.max(0.2, Math.abs(dx));
+        break;
+      case Sprite.Edge.RIGHT:
+        dx = -Math.max(0.2, Math.abs(dx));
+        break;
+      case Sprite.Edge.TOP:
+        dy = -Math.max(0.2, Math.abs(dy));
+        break;
+      case Sprite.Edge.BOTTOM:
+        dy = Math.max(0.2, Math.abs(dy));
+        break;
+    }
+    this.direction = this.radToScratch(Math.atan2(dy, dx));
+    const { x, y } = this.keepInFence(this.x, this.y);
+    this.goto(x, y);
```

Comment:
Oh! Quick addition. Is there a reason we need to separate the call to `this.goto(x, y)` from `this.keepInFence()`? It's not a huge deal I imagine, especially as `ifOnEdgeBounce` should realistically only be called once per relevant sprite per frame, but it does create an extra short-lived `{x, y}`, and we could optimize it out by putting the call to `this.goto()` at the end of the `keepInFence` method. I don't see another use for calling `keepInFence` separately (where you would want to do something *else* with the `{x, y}` coordinates *instead* of repositioning the sprite).

Reply:
Yeah, I can't see a huge reason for this either way.

I think the original reason why this was [separated in the Scratch-VM](https://github.com/LLK/scratch-vm/search?q=keepinfence) was so that the rendering code (which understands `keepInFence`) could be completely separated from the movement code (which handles the `goto`).  However, we have them all in the same file here, so I don't see as huge of a need to separate them.

The only other reason I could see for keeping them separated like this is just for code compatibility / re-use from the Scratch-VM (so that the two functions named the same thing have the same behavior / side-effects), but it's not a huge deal either way in my mind.  I'm fine going either way.

Reply:
I'd personally vote for moving `goto` and perhaps renaming `keepInFence` to `positionInFence` to indicate that it's not *exactly* the same behavior as in scratch-vm (and identifies what the difference is—though I'd argue the scratch-vm method should be renamed since "keep" sounds like it has side effects even though the function there does not!). Maybe others would approach this differently though :)

Reply:
Implemented in 5186824